server/src/main/java/org/apache/druid/metadata/SQLMetadataSupervisorManager.java

```diff
@@ -249,6 +252,34 @@ public Void withHandle(Handle handle) throws Exception
     );
   }
 
+  @Override
+  public int removeTerminatedSupervisorsOlderThan(long timestamp)
+  {
+    int removedCount = 0;
+    DateTime dateTime = DateTimes.utc(timestamp);
+    Map<String, SupervisorSpec> supervisors = getLatest();
+    for (Map.Entry<String, SupervisorSpec> supervisor : supervisors.entrySet()) {
+      final SupervisorSpec spec = supervisor.getValue();
+      if (spec instanceof NoopSupervisorSpec) {
```

Comment:
does this mean that it is a terminated supervisor? can you add a comment clarifying this bit

Reply:
Yep. Terminated supervisor will have it's latest supervisorSpec as NoopSupervisorSpec. (Basically NoopSupervisorSpec is used as a tombstone indicator)